Pulsed ODMR and XY8 sequences for NV centers

NicolasStaudenmaier opened the pull request:
This PR implements the sequences of pulsed ODMR and XY8 for NV centers. Pulsed ODMR is a basic and widely used measurement to detect the spin resonance frequency. When calibrated well, it usually gives much better and precise results than cw ODMR. 
Furthermore, the dynamical decoupling sequence XY8-N of order N is added. The sequence can be performed in two different modes. One is sweeping the interpulse spacing (tau) between the pi-pulses and the other is sweeping the order N (number of repetitions) of the XY8 block. 
The `README.md` file is updated accordingly and some file names are changed to ensure consistent naming with the general structure in this repo. 

The XY8 tau sweep sequence shows no problems with unintentional gaps within the XY8 sequence during runtime. However, for the XY8 order sweep a gap may occur because a dynamic QUA `for_()` loop is required to cycle through the orders. I found that with the implementation in `09c_xy8_order.py` a single gap is evaluated before entering the loop. Importantly, the command structure as it is reduces the number of gaps to one. With other structures, gaps inside the loop were determined. 
The exact duration of the gap depends on the sequence parameters and I determined something between no gap and 10 clock cycles. It can be compensated by adding the gap time between the end of the XY8-block and the final pi/2-pulse. When calibrated properly, this operation does not affect the physics of the sequence. 

Proper implementation is verified with the waveform report in simulated mode on the OPX+ in the CS_1 rack.

TheoLaudatQM commented on Quantum-Control-Applications/Optically addressable spin qubits/NV center in a confocal setup/09b_xy8_tau.py:
```diff
@@ -0,0 +1,217 @@
+#######################
+# Simulate or execute #
+#######################
+simulate = True
```
I would set the default to False so that you don't have to always toggle it when running the script during an installation

TheoLaudatQM commented on Quantum-Control-Applications/Optically addressable spin qubits/NV center in a confocal setup/09c_xy8_order.py:
```diff
@@ -0,0 +1,211 @@
+#######################
+# Simulate or execute #
+#######################
+simulate = True
```
I would set the default to False so that you don't have to always toggle it when running the script during an installation

NicolasStaudenmaier commented on Quantum-Control-Applications/Optically addressable spin qubits/NV center in a confocal setup/05b_pulsed_odmr.py:
```diff
@@ -0,0 +1,145 @@
+#######################
+# Simulate or execute #
+#######################
+simulate = True
```
Totally. Also to be consistent.

NicolasStaudenmaier commented on Quantum-Control-Applications/Optically addressable spin qubits/NV center in a confocal setup/05b_pulsed_odmr.py:
```diff
@@ -0,0 +1,145 @@
+Prerequisites:
+    - Ensure calibration of the different delays in the system (calibrate_delays).
+    - Update the different delays in the configuration
+    - Setting the pi pulse amplitude and duration in the configuration
```
In my experience you do both Rabi and pulsed ODMR initially with some good guess as you know the order of the params in your setup. For Rabi you have to "guess" the frequency, for pulsed ODMR it is the duration/amplitude. Then you update them one another. 
But I can also remove that line as in the list of measurements Rabi follows after this.

NicolasStaudenmaier commented:
Thank you for review, @TheoLaudatQM. I have addressed your comments. 
I will further work on the XY8 wrapper after discussing with Daniel. I will notify here, when it is ready for review again.

TheoLaudatQM commented on Quantum-Control-Applications/Optically addressable spin qubits/NV center in a confocal setup/09b_xy8_tau.py:
```diff
@@ -130,6 +130,7 @@ def xy8_block(tau):  # A single XY8 block
 
     with for_(n, 0, n < n_avg, n + 1):
         with for_each_((tau_spacing, tau_half_spacing), (tau_vec_spacing, tau_half_vec_spacing)):
+            wait(4)
```
What is this for?

TheoLaudatQM replied:
Yes please :)

TheoLaudatQM commented on Quantum-Control-Applications/Optically addressable spin qubits/NV center in a confocal setup/09c_xy8_order.py:
```diff
@@ -43,7 +43,7 @@
 tau = 2 * 50
 order_vec = np.arange(1, 21, 1, dtype=int)  # order vector for varying the order n of the XY8-n measurement
-n_avg = 1_000
+n_avg = 1_000_000
```
Is 1000000 averages really needed as the default value?
I usually prefer to have a smaller number of averages to ensure that the data is correct quickly, before increasing it to improve the SNR, but I leave it to you.

TheoLaudatQM reviewed (changes requested):
Nice, this is an interesting way of performing the readout!
I didn't test it, so I am trusting you that it compiles and simulates properly.
The only comment is that the docstrings at the beginning of the scripts should be updated to reflect the changes.
